### PR TITLE
feat: Add `--abort` and `--user_range` flags

### PR DESCRIPTION
**Pull request by jrconlin**
## Description

* --abort stops copying BSO records after N instances.
* --user_range limits copy to offset:limit users.
* optionally sorts users by fxa_uid via `--sort_users` flag
* collapses previous pushes and syncs with master branch

## Testing

* configure local environment using the steps provided in the `tools/README.md` file
* select a contiguous range of users specified in the given node starting from a given offset (e.g. if a node contains 100 users, you wish to copy 20 users, and you wish to skip the first 10 users, use a `user_range=10:20`. Verify users being copied fit the specified range.
* for a given category, select a subset of records you wish to copy over. Other records are dropped. (e.g. if a user has 100 history items and you wish to test that only 10 are copied over, use `abort=history:10`). App reports rows added or skipped.
* Can use `--dryrun` which will not write data to spanner, but act as if it did. 

## Issue(s)

Issue #407

### CODE REVIEW

File: tools/user_migration/migrate_node.py [hunk 1]
```diff
@@ -451,34 +459,52 @@ def spanner_transact_bso(transaction, data, fxa_kid, fxa_uid, args):
...
+                try:
+                    (fxa_kid, fxa_uid) = fxa.get(user)
+                    users.append((user, fxa_kid, fxa_uid))
+                except TypeError:
+                    logging.error("⚠️User not found in tokenserver data: {} ".format(
+                        user
+                    ))
+            users.sort(key=lambda tup: tup[2])
+        except Exception as ex:
+            import pdb; pdb.set_trace()
```
Comment (tublitzed) [inline tools/user_migration/migrate_node.py]: ✂️ ?

Comment (jrconlin) [reply tools/user_migration/migrate_node.py]: well, it's in an exception, and the script is meant to be hand-run. I'd rather it dump to an examinable state at that point.

**Comment by tublitzed** [comment]
Would it be possible to isolate the rust migration script from this PR so we can get this version merged and review that separately?

**Comment by tublitzed** [comment]
@jrconlin can you update the testing instructions here? What specifically are we supposed to run to test this?

**Comment by jrconlin** [comment]
@tublitzed comment updated.

**Pull request merged**